test/int/cluster_test.go

```diff
@@ -122,6 +122,43 @@ var _ = Describe("AtlasCluster", func() {
 	}
 
 	Describe("Create/Update the cluster", func() {
+		It("Should fail, then be fixed", func() {
+			expectedCluster := testAtlasCluster(namespace.Name, "test-cluster", createdProject.Name)
+			expectedCluster.Spec.Name = ""
+
+			By(fmt.Sprintf("Creating the Cluster %s with invalid parameters", kube.ObjectKeyFromObject(expectedCluster)), func() {
+				createdCluster.ObjectMeta = expectedCluster.ObjectMeta
+				Expect(k8sClient.Create(context.Background(), expectedCluster)).ToNot(HaveOccurred())
+
+				Eventually(
+					testutil.WaitFor(
+						k8sClient,
+						createdCluster,
+						status.
+							FalseCondition(status.ClusterReadyType).
+							WithReason(string(workflow.Internal)).
```

Comment:
[q]  I would expect this to be `ClusterNotCreatedInAtlas` instead of `Internal` or am I missing something?

Reply:
Ok, I know the answer. It's not the `create` request that fails, but the `client.Clusters.Get(ctx, project.Status.ID, cluster.Spec.Name)` and this results in the internal error.

Can we please add the comment here to explain this as logically it seems that the creation should fail, not get